src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -61,7 +59,7 @@ public class DurableTaskExtension :
         private readonly bool isOptionsConfigured;
         private IDurabilityProviderFactory durabilityProviderFactory;
         private INameResolver nameResolver;
-        private DurabilityProvider defaultDurabilityProvider;
+        private Lazy<DurabilityProvider> defaultDurabilityProvider;

```

Comment:
I'm not sure this second layer of `Lazy` is better or worse than my approach of changing the order of when we resolve app settings into the options vs when we get our durability provider. 

Your approach is probably a bit less brittle than mine, but I am afraid it would be harder to debug if it broke for some reason. Mine is a bit more explicit in the intention that we need to resolve app settings before getting our durability provider.